src/neuroconv/tools/spikeinterface/spikeinterface.py

```diff
@@ -656,18 +656,27 @@ def add_electrical_series(
         data=H5DataIO(data=ephys_data_iterator, compression=compression, compression_opts=compression_opts)
     )
 
-    # Timestamps vs rate
-    timestamps = recording.get_times(segment_index=segment_index)
-    rate = calculate_regular_series_rate(series=timestamps)  # Returns None if it is not regular
-    starting_time = starting_time if starting_time is not None else 0
+    # Now we decide whether to store the timestamps as a regular series or as an irregular series.
+    if recording.has_time_vector(segment_index=segment_index):
+        # First we check if the recording has a a time vector to avoid creating artificial timestamps
+        timestamps = recording.get_times(segment_index=segment_index)
+        rate = calculate_regular_series_rate(series=timestamps)  # Returns None if it is not regular
+        recording_t_start = timestamps[0]
+    else:
+        rate = recording.get_sampling_frequency()
+        recording_t_start = recording._recording_segments[segment_index].t_start
+        recording_t_start = 0 if recording_t_start is None else recording_t_start
```

Comment:
```suggestion
        recording_t_start = recording._recording_segments[segment_index].t_start or 0
```

if I understand correctly, this simpler version should work

Reply:
Yes, this is better!